examples/rbfe_single.py

```diff
@@ -183,8 +191,7 @@ def run_epoch(ff, mol_a, mol_b, core):
 
     cmd_args = parser.parse_args()
 
-    multiprocessing.set_start_method('spawn') # CUDA runtime is not forkable
```

Comment:
@maxentile You should not remove this. This also affects the launch mode of ProcessPoolExecutors. It will default to 'fork' which is incompatible with CUDA, and as it turns out, jax.

Reply:
Ahh, thank you for catching this! Would this be okay to move into the `ProcessPoolClient` or `CUDAPoolClient` constructor then? (I had mistakenly assumed it was safe to remove from this file because there was no longer a direct call to any other multiprocessing function defined in the script.)

Reply:
I think that's fine, an less intrusive alternative would be to assert in the constructor of the PoolClients:
```
multiprocessing.get_start_method() == 'spawn'
```
This way the end-user is still conscientious of the start mode. Generally speaking, library code that set globally hidden singletons is a bad idea, and these types of decisions (like setting jax's precision) should be done immediately and explicitly at the start of the application script. Trying to debug this later on (because some other code might assume forking) will be a goddamn nightmare. Some other fun bits: on windows the default start mode is actually already 'spawn', but on linux for some reason the default mode 'fork'.